src/app/three-examples/objects/primitive/torusmesh.directive.ts

```diff
@@ -0,0 +1,48 @@
+import { Directive, Input, AfterViewInit, forwardRef } from '@angular/core';
+import * as THREE from 'three';
+import { AbstractMesh } from '../abstract-mesh-3d';
+import { AbstractObject3D } from '../abstract-object-3d';
+
+@Directive({
+  selector: 'three-torusmesh',
+  providers: [{ provide: AbstractObject3D, useExisting: forwardRef(() => TorusmeshDirective) }]
+})
+export class TorusmeshDirective extends AbstractMesh {
+  //  - Radius of the torus, from the center of the torus to the center of the tube. Default is 1.
+  @Input()
+  radius: number; 
+  //  — Radius of the tube. Default is 0.4.
+  @Input()
+  tube: number; 
+  @Input()
+  radialSegments: number; //  — Default is 8
+  @Input()
+  tubularSegments: number; //  — Default is 6.
+  @Input()
+  arc: number; // — Central angle. Default is Math.PI * 2. 
+
+  constructor() { 
+    super();
+    console.log('TorusmeshDirective.constructor');
+  }
+
+  protected newObject3DInstance(): THREE.Mesh {
+    this.radius*=1;
```

Comment:
I'd prefer spacing `this.radius *= 1`. However, the more important note is: is `*= 1` a tricky workaround for when the input variable has not been set? Why not specify the default directly as `@Input() tube = 1` (note the type being left out).

Reply:
It is a tricky WA to convert the input to number. I don't understand but if I leave it out than the attributes from html are not converted to number...

Reply:
Maybe because they are specified with quotes in the HTML template?

Reply:
After adding ' around the parameters in the html template I get the same problem. (it works only with the WA...)

Reply:
By adding " around the parameters in html the same again... :\